### PR TITLE
leak caught

### PR DESCRIPTION
**Pull request by forbidden-arts**
parseHttpRequest was most of the issue. put inside a catch block and delete the request when it fails. it rethrows the same error it received.

### CODE REVIEW

File: src/HttpRequestParser.cpp [hunk 1]
```diff
@@ -31,6 +33,12 @@ HttpRequest *HttpRequestParser::parseHttpRequest(std::string requestInput, Serve
 	}
 	request->setHost(request->getHeader("Host"));
 	return request;
+	}
+	catch (...)
+	{
+		delete request;
+		throw;
```
Comment (merituulie) [inline src/HttpRequestParser.cpp:40]: throwing what? does this shut down the program? that shouldnt happen

Comment (merituulie) [reply src/HttpRequestParser.cpp:40]: we also need to add the specific exception to be catched here.. rather multiple catches that catchs different exceptions and return after the try catch block.

Comment (merituulie) [reply src/HttpRequestParser.cpp:40]: Also, the request will be deleted in the error response thing in the server handler so the leak is not the request itself... in anycase it speaks about some deeper issue I think

Comment (forbidden-arts) [reply src/HttpRequestParser.cpp:40]: so the ellipses is a catch-all, and the empty throw will throw the same error it received

**Review by merituulie — APPROVED** [review]

**Review by rosamakinen — APPROVED** [review]
looks good to me

**Pull request merged**